tests/testthat/test-translate-teradata.r

```diff
@@ -0,0 +1,150 @@
+context("translate-teradata")
+
+# Teradata base_scalar conversions -----------------------------------------
+
+test_that("as.numeric() translated to NUMERIC ", {
+  expect_equivalent(
+    translate_sql(as.numeric(field_name), con = simulate_teradata()),
+    sql("CAST(`field_name` AS NUMERIC)")
+  )
+})
+
+test_that("as.double() translated to NUMERIC ", {
+  expect_equivalent(
+    translate_sql(as.double(field_name), con = simulate_teradata()),
+    sql("CAST(`field_name` AS NUMERIC)")
+  )
+})
+
+test_that("as.character() translate to VARCHAR(MAX) ", {
+  expect_equivalent(
+  translate_sql(as.character(field_name), con = simulate_teradata()),
+    sql("CAST(`field_name` AS VARCHAR(MAX))")
+  )
+})
+test_that("log10() translates to LOG ", {
+  expect_equivalent(
+    translate_sql(log10(field_name), con = simulate_teradata()),
+    sql("LOG(`field_name`)")
+  )
+})
+
+test_that("log() translates to ln ", {
+  expect_equivalent(
+    translate_sql(log(field_name), con = simulate_teradata()),
+    sql("ln(`field_name`)")
+  )
+})
+
+test_that("log() with a different base translates to formula ", {
+  expect_equivalent(
+    translate_sql(log(field_name, base = 2) , con = simulate_teradata()),
+    sql("log(`field_name`) / log(2.0)")
+  )
+})
+
+
+test_that("ceiling() translates to CEILING ", {
+  expect_equivalent(
+    translate_sql(ceiling(field_name), con = simulate_teradata()),
+    sql("CEILING(`field_name`)")
+  )
+})
+
+test_that("paste() returns error message", {
+  expect_error(
+    translate_sql(paste(field_name),
+                  window = FALSE,
+                  con = simulate_teradata()),
+    "PASTE\\(\\) is not supported in this SQL variant, try PASTE0\\(\\) instead"
+  )
+})
+
+# Teradata base_agg conversions -----------------------------------------
+
+test_that("sd() translates to STDEV ", {
```

Comment:
Can you please bundle these all into one test, writing a function as needed to reduce duplication?